packages/flutter_tools/lib/src/debug_adapters/flutter_adapter.dart

```diff
@@ -85,30 +85,55 @@ class FlutterDebugAdapter extends DartDebugAdapter<FlutterLaunchRequestArguments
 
   /// Whether or not the user requested debugging be enabled.
   ///
-  /// debug/noDebug here refers to the DAP "debug" mode and not the Flutter
-  /// debug mode (vs Profile/Release). It is provided by the client editor based
-  /// on whether a user chooses to "Run" or "Debug" their app.
+  /// For debugging to be enabled, the user must have chosen "Debug" (and not
+  /// "Run") in the editor (which maps to the DAP `noDebug` field) _and_ must
+  /// not have requested to run in Profile or Release mode. Profile/Release
+  /// modes will always disable debugging.
   ///
-  /// This is always enabled for attach requests, but can be disabled for launch
-  /// requests via DAP's `noDebug` flag. If `noDebug` is not provided, will
-  /// default to debugging.
+  /// This is always `true` enabled for attach requests.
```

Comment:
nit. this sentence is awkward. should it be "This is always `true` for attach requests."?

Reply:
Oops! Awkward is an understatement 😄  Fixed!